src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+.calendar {
+  margin: auto;
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: $day-gap;
+  max-width: ($day-size * 7) + ($day-gap * 6);
+
+  @each $day, $count in $days {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($day-size + $day-gap) * $count;
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} :nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    background-color: $day-bg-color;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid black;
+    position: relative;
+    cursor: pointer;
+    transition-duration: 0.5s;
+
+    &:hover {
+      background-color: $day-hover-color;
+      transform: translateY(-20px);
```

Comment:
The transform property should be transitioned for a smoother hover effect. Add 'transform' to the 'transition-property' to comply with the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules).